### DISCUSSION TITLE
Astro + Fastify adapter (instead of node + custom express)

### DISCUSSION THREAD

**Discussion by edv-ns** [withastro/roadmap#463] [Proposal]
### Body

# Summary

The **Astro+Fastify** combination is better than Astro+Express, and I am sure many developers would like to be able easily integrate it into their projects.

# Background & Motivation

As seen in benchmarks (https://www.fastify.io/benchmarks/) the **Fastify** is much faster (4x) option than good old **Express**. 

![image](https://user-images.githubusercontent.com/73230019/216361360-f444d07b-5a96-487c-ac3a-6fbad9c4ff80.png)

@matthewp did a good job in making his package [@matthew/astro-fastify](https://github.com/matthewp/astro-fastify)

His package benefits:
- Easy-integratable
- Unlike most adapters, works in dev mode !

Since the adapter works with Vite and goes through internals of Astro, it had to patch some bits to support the Astro.cookies API and probably other native functionality in future. That may be a reason why it could/should be included into official monorepo.

# Goals

- Make `@matthew/astro-fastify` as official Astro+Fastify adapter
- Add documentation

**Comment by matthewp** [comment]
Just to clarify, that integration does not patch anything. It works the same way as all of the integrations. I think the thing you are pointing out is that it conditionally checks for an API. It does that to avoid breaking backwards compatibility with older Astro versions.

**Comment by florian-lefebvre** [comment]
I don't think we will make this adapter official. If you need dev support for the node adapter, follow #344. If you want to use fastify with the node adapter in prod, use middleware mode: https://docs.astro.build/en/guides/integrations-guide/node/#mode
